### DISCUSSION TITLE
Should strongswan be restarted on system date/time change

### DISCUSSION THREAD

**Discussion by nikhilbhandari123** [strongswan/strongswan#2213] [General]
When system date or time is changed on Ubuntu based systems, should strongswan be restarted ? I know that strongswan uses CLOCK_MONOTONIC, so if I do not restart strongswan, what will happen ?

**Comment by tobiasbrunner** [comment]
No, I don't think you have to restart it.

> so if I do not restart strongswan, what will happen ?

With regards to what?

**Comment by nikhilbhandari123** [comment]
> With regards to what?
Will it cause some problems with rekeying or when any certificate expiry happens. Basically, are these statements valid:
If StrongSwan relies on certificates with validity periods based on system time, a significant system time change could lead to inconsistencies in certificate validation.
Time-based mechanisms such as rekeying of IPsec SAs or expiration of IKE exchanges may not align correctly with the system's notion of time, potentially causing operational issues.

**Reply by tobiasbrunner** [reply]
> If StrongSwan relies on certificates with validity periods based on system time, a significant system time change could lead to inconsistencies in certificate validation.

Correct, certificates are not re-evaluated until the peer re-authenticates.

> Time-based mechanisms such as rekeying of IPsec SAs or expiration of IKE exchanges may not align correctly with the system's notion of time, potentially causing operational issues.

No, these things use monotonic time, so they are not affected.
